GCRCatalogs/dc2_truth_match.py

```diff
@@ -0,0 +1,87 @@
+"""
+Reader for truth-match catalogs persisted as parquet files and partitioned in tracts.
+"""
+import numpy as np
+import astropy.units as u
+
+from .dc2_dm_catalog import DC2DMTractCatalog
+
+__all__ = ["DC2TruthMatchCatalog"]
+
+
+def _flux_to_mag(flux):
+    with np.errstate(divide="ignore"):
+        return (flux * u.nJy).to_value(u.ABmag)  # pylint: disable=no-member
+
+
+class DC2TruthMatchCatalog(DC2DMTractCatalog):
+    r"""
+    DC2 Truth-Match (parquet) Catalog reader
+
+    Presents tables exactly as they are defined in the files (no aliases,
+    no derived quantities)
+
+    Parameters
+    ----------
+    base_dir          (str): Directory of data files being served, required
+    filename_pattern  (str): The optional regex pattern of served data files
+    as_object_addon  (bool): If set, return rows in the the same row order as object catalog
+    as_truth_table   (bool): If set, remove duplicated truth rows
+    """
+
+    def _subclass_init(self, **kwargs):
+
+        super()._subclass_init(**dict(kwargs, is_dpdd=True))  # set is_dpdd=True to obtain bare modifiers
+
+        self._as_object_addon = bool(kwargs.get("as_object_addon"))
+        self._as_truth_table = bool(kwargs.get("as_truth_table"))
+        if self._as_object_addon and self._as_truth_table:
+            raise ValueError("Reader options `as_object_addon` and `as_truth_table` cannot both be set to True.")
+
+        flux_cols = [k for k in self._quantity_modifiers if k.startswith("flux_")]
+        for col in flux_cols:
+            self._quantity_modifiers["mag_" + col.partition("_")[2]] = (_flux_to_mag, col)
+
+        if self._as_object_addon:
+            no_postfix = ("truth_type", "is_unique_truth_entry", "match_sep", "match_objectId")
+            self._quantity_modifiers = {
+                (k + ("" if k in no_postfix else "_truth")): (v or k) for k, v in self._quantity_modifiers.items()
+            }
+
+    def _obtain_native_data_dict(self, native_quantities_needed, native_quantity_getter):
+        """
+        Overloading this so that we can query the database backend
```

Comment:
docstring (left over from original truth reader for sqlite dbs?) needs to be updated, to say something about adding other quantities and special handling as appropriate for the mode (as addon or as truth table) of use.